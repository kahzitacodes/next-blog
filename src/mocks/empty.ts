import { EmptyProps } from '@/components/empty'

export const emptyMock: EmptyProps = {
  title: 'Oops!',
  description: 'Page not found'
}
