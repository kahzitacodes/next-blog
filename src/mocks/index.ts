export * from './navigation'
export * from './profile'
export * from './social-links'
export * from './post'
export * from './pre'
export * from './pagination'
