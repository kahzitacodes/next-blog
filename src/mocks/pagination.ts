export const paginationMock = {
  currentPage: 1,
  totalPages: 10
}
