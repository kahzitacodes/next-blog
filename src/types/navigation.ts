import * as LucideIcons from 'lucide-react'

export type NavItem = {
  title: string
  href: string
  icon?: keyof typeof LucideIcons
}
