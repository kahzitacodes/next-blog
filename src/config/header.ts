import { HeaderNavItems } from '@/types'

export const headerConfig: HeaderNavItems = [
  {
    title: 'Home',
    href: '/'
  },
  {
    title: 'Artigos',
    href: '/articles'
  },
  {
    title: 'Sobre mim',
    href: '/about'
  }
]
