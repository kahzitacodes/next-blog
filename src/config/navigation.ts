import { NavItem } from '@/types'

export const navigationConfig: NavItem[] = [
  {
    title: 'Home',
    href: '/'
  },
  {
    title: 'Posts',
    href: '/posts'
  },
  {
    title: 'About me',
    href: '/about'
  }
]
