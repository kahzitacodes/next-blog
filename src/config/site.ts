import { Site } from '@/types'

export const siteConfig: Site = {
  name: 'Karine Pereira',
  avatar: '/assets/images/avatar.jpeg',
  description: 'Blog onde falo sobre arte, design e tecnologia',
  title: 'Frontend developer and Interface designer',
  url: 'http://meusite.com.br',
  subtitle:
    'Welcome to my blog! My name is Karine, I love technology, and everyday I try to learn something new.',
  links: [
    {
      title: 'Github',
      href: 'http://github.com',
      icon: 'Github'
    },
    {
      title: 'LinkedIn',
      href: 'http://linkedin.com',
      icon: 'Linkedin'
    }
  ]
}
