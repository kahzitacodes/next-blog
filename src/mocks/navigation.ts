import { NavItem } from '@/types'

export const navigationMock: NavItem[] = [
  {
    title: 'Home',
    href: '/'
  },
  {
    title: 'About',
    href: '/about'
  },
  {
    title: 'Contact',
    href: '/contact'
  }
]
