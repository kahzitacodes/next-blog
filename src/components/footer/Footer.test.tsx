import { render, screen } from '@testing-library/react'
import { navigationMock } from '@/mocks/navigation'
import { Footer } from '.'
import { siteConfig } from '@/config'

describe('<Footer />', () => {
  it('should render successfully', () => {
    const { container } = render(<Footer items={navigationMock} />)

    expect(screen.getByText(siteConfig.title)).toBeInTheDocument()
    expect(screen.getByTestId(/logo/i)).toBeInTheDocument()
    expect(screen.getByRole('heading', { name: /Social/i })).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: /Sitemap/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
