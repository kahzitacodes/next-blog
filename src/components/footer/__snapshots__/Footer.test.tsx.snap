// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render successfully 1`] = `
<div
  class="mx-auto w-full max-w-6xl px-6 lg:max-w-6xl"
>
  <footer
    class="flex flex-col py-8 border-t border-gray-700"
  >
    <div
      class="flex w-full justify-between pb-10"
    >
      <section
        class="flex max-w-[12rem] flex-col gap-6 md:max-w-[36rem]"
        role="contentinfo"
      >
        <a
          class=""
          href="/"
        >
          <img
            alt="Logo"
            data-nimg="1"
            data-testid="logo"
            decoding="async"
            height="21"
            loading="lazy"
            src="/assets/images/logo.svg"
            style="color: transparent;"
            width="28"
          />
        </a>
        <p>
          Frontend developer and Interface designer
        </p>
      </section>
      <section
        class="md:max-w-[36rem] flex max-w-[8rem] flex-col gap-6 sm:max-w-lg"
        role="contentinfo"
      >
        <h5
          class="font-semibold"
        >
          Social
        </h5>
        <ul
          class="flex gap-3 flex-col"
        >
          <li
            class="flex gap-2 font-light text-medium relative [&>a]:hover:text-link [&>a]:transition-colors [&>a]:duration-200 [&>svg]:hover:text-link"
          >
            <a
              class="flex justify-center gap-3"
              href="http://github.com"
            >
              <span
                class="flex w-6"
              >
                <svg
                  class="lucide lucide-github"
                  data-testid="Github"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M15 22v-4a4.8 4.8 0 0 0-1-3.5c3 0 6-2 6-5.5.08-1.25-.27-2.48-1-3.5.28-1.15.28-2.35 0-3.5 0 0-1 0-3 1.5-2.64-.5-5.36-.5-8 0C6 2 5 2 5 2c-.3 1.15-.3 2.35 0 3.5A5.403 5.403 0 0 0 4 9c0 3.5 3 5.5 6 5.5-.39.49-.68 1.05-.85 1.65-.17.6-.22 1.23-.15 1.85v4"
                  />
                  <path
                    d="M9 18c-4.51 2-5-2-7-2"
                  />
                </svg>
              </span>
              Github
            </a>
          </li>
          <li
            class="flex gap-2 font-light text-medium relative [&>a]:hover:text-link [&>a]:transition-colors [&>a]:duration-200 [&>svg]:hover:text-link"
          >
            <a
              class="flex justify-center gap-3"
              href="http://linkedin.com"
            >
              <span
                class="flex w-6"
              >
                <svg
                  class="lucide lucide-linkedin"
                  data-testid="Linkedin"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"
                  />
                  <rect
                    height="12"
                    width="4"
                    x="2"
                    y="9"
                  />
                  <circle
                    cx="4"
                    cy="4"
                    r="2"
                  />
                </svg>
              </span>
              LinkedIn
            </a>
          </li>
        </ul>
      </section>
      <section
        class="md:max-w-[36rem] flex max-w-[8rem] flex-col gap-6 sm:max-w-lg"
        role="contentinfo"
      >
        <h5
          class="font-semibold"
        >
          Sitemap
        </h5>
        <ul
          class="flex gap-3 flex-col"
        >
          <li
            class="flex gap-2 font-light text-medium relative [&>a]:hover:text-link [&>a]:transition-colors [&>a]:duration-200 [&>svg]:hover:text-link"
          >
            <a
              class="flex justify-center gap-3"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="flex gap-2 font-light text-medium relative [&>a]:hover:text-link [&>a]:transition-colors [&>a]:duration-200 [&>svg]:hover:text-link"
          >
            <a
              class="flex justify-center gap-3"
              href="/about"
            >
              About
            </a>
          </li>
          <li
            class="flex gap-2 font-light text-medium relative [&>a]:hover:text-link [&>a]:transition-colors [&>a]:duration-200 [&>svg]:hover:text-link"
          >
            <a
              class="flex justify-center gap-3"
              href="/contact"
            >
              Contact
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      class="mt-3 flex w-full items-center justify-center text-gray-500"
    >
      <p>
        Todos os direitos reseverdos Â© Karine Pereira 
        2025
      </p>
    </div>
  </footer>
</div>
`;
